### ISSUE TITLE
webp images are not showing on wamp/xamp

### ISSUE THREAD

**Issue by AbleOne** [rosell-dk/webp-convert#77]
Hi there!

I'm trying to run your plugin on wamp and it's ok, because I can see that converter works here - [http://take.ms/l3miu](http://take.ms/l3miu), and here from response header: [http://take.ms/L2A0E](http://take.ms/L2A0E)

but I have several problems: 
1. images are not displyed [http://take.ms/AflZY](http://take.ms/AflZY) while they served properly [http://take.ms/l63GC](http://take.ms/l63GC) can you please help me ton understand how to show them
2. GD extension worked for me, but I have message that "WebP Express cannot save test conversion..." [http://take.ms/rAUUF](http://take.ms/rAUUF) while I pressing "Save Settings" (even without force .htaccess) and see that test is successfully saved [http://take.ms/kugVN](http://take.ms/kugVN)
Clicking "Convert test image (show debug)" shows that everyting has success [http://take.ms/cGbcg](http://take.ms/cGbcg), but if I click "Convert test image" - I see only some little square with white border  [http://take.ms/2O23L](http://take.ms/2O23L) while it should look like this [http://take.ms/atLIU](http://take.ms/atLIU) I guess)

wamp:
apache 2.4.35
php 5.6.28

folders in wp-contet and in general all wp folders has all access (read/write) permissions (not "only read")

Please help me) I've almost done this))

**Comment by rosell-dk** [comment]
I haven't done any testing on Windows, as I do not have a licence, and aren't inclined to spend money on a license just for this project. Additionally, I haven't taken any measures to make it work on Windows. So it is no surprise to me that it doesn't work. What surprises me is that from what you write, it seems to *almost* work. The statuses you report seems contradicting, but I guess it must be because there is some code in this library, or in WebP Express that fails on Wamp.

**Comment by AbleOne** [comment]
Hi thanks for looking this through!)
I have licensed Win10 with my notebook)
Ok, let's drop windows) I've set up test wordpress instance and installed webp express there. In plugin  settings it seems that everything is ok [http://take.ms/rLZoo7](http://take.ms/rLZoo7). But when I try to open test.jpg ("Convert test image (show debug)" link) I don't see report, I see just only just test.jpg (not webp) [http://take.ms/OzpXn](http://take.ms/OzpXn)
As well as on the front I still see just jpegs [http://take.ms/iDycC](http://take.ms/iDycC). Could you please help me to understand what I'm doing wrong) I really need this))
I've sent link on wordpress instance and credentials to /wp-admin on your email that you mentioned at [https://www.bitwise-it.dk/contact](https://www.bitwise-it.dk/contact)

**Comment by rosell-dk** [comment]
Thanks for credentials. In the new setup, the redirect aren't working. Inspecting the response headers, I see: "Server: openresty". I do not know that server, but from its [homepage](https://openresty.org/en/), it seems it is based on Nginx. WebP Express is currently only able to generate Apache rules (compatible with LiteServer). Nginx rules are however on the roadmap. 

Unless you have altered location of plugins or wp-content, you should be able to get the redirection working by inserting these rules in your Nginx configuration (nginx.conf):

```
if ($http_accept ~* "webp"){
rewrite ^/(.*).(jpe?g|png)$ /wp-content/plugins/webp-express/wod/webp-on-demand.php?source=$document_root$request_uri&wp-content=wp-content&%1 break;
}
```

If that doesn't work, try looking at the links I posted in this issue:
https://wordpress.org/support/topic/nginx-rewrite-rules-4/#post-10856041

**Comment by rosell-dk** [comment]
Back to the Wamp-issue:
It puzzled me that the test button works, but the ?debug link fails. Today however, I ran into exactly such a situation. It can happen because the test-buttons calls webp-convert library directly (composer autoload), while the webp-on-demand.php (which images are redirected to, including ?debug) runs on two files, "webp-on-demand-1.inc" and "webp-on-demand-2.inc", which are the webp-convert library cut in two. The first part, "webp-on-demand-1.inc" takes care of serving cached images quickly and loads the second (and much larger) part, if an image needs to be converted.
So it is not exactly the same files, they are running on. If they don't behave alike, there can be two causes. Either I did not run the build command before submitting, or there is a bug in my script that builds these files.

Long story short, you can try to run with autoload instead. 
1. Edit webp-express/wod/webp-on-demand.php
2. Comment out "require '../vendor/rosell-dk/webp-convert/build/webp-on-demand-1.inc'" (line 4)
3. Uncomment "require '../vendor/autoload.php';" (line 5)
4. Comment out "$options['require-for-conversion'] = 'webp-on-demand-2.inc';" (~line 21)

**Comment by rosell-dk** [comment]
closed due to inactivity
